docs/src/main/asciidoc/security-proactive-authentication.adoc

```diff
@@ -13,36 +13,44 @@ Proactive authentication is enabled in Quarkus by default. This means that if an
+For blocking applications, this is not an issue.
+However, if you have disabled authentication in a reactive application, this will fail because you cannot do blocking operations on the I/O thread.
 To work around this, you need to `@Inject` an instance of `io.quarkus.security.identity.CurrentIdentityAssociation` and call the `Uni<SecurityIdentity> getDeferredIdentity();` method.
-You can then subscribe to the resulting `Uni` and will be notified when authentication is complete and the identity is available.
+Then, you can subscribe to the resulting `Uni` and will be notified when authentication is complete and the identity is available.
 
-NOTE: It's still possible to access the `SecurityIdentity` synchronously with `public SecurityIdentity getIdentity()` in the xref:resteasy-reactive.adoc[RESTEasy Reactive] from endpoints annotated with `@RolesAllowed`, `@Authenticated`, or with respective configuration authorization checks as authentication has already happened.
-The same is also valid for the xref:reactive-routes.adoc[Reactive routes] if a route response is synchronous.
+[NOTE]
+====
+You can still access `SecurityIdentity` synchronously with `public SecurityIdentity getIdentity()` in xref:resteasy-reactive.adoc[RESTEasy Reactive] from endpoints that are annotated with `@RolesAllowed`, `@Authenticated`, or with respective configuration authorization checks because authentication has already happened.
+The same is also valid for xref:reactive-routes.adoc[Reactive routes] if a route response is synchronous.
+====
 
-xref:security-authorization.adoc#standard-security-annotations[Standard security annotations] on CDI beans are not supported on IO thread if a non-void secured method returns a value synchronously and proactive authentication is disabled, as they need to access the `SecurityIdentity`.
-In the example below, we have defined `HelloResource` and `HelloService`. It's easy to see that any GET request to `/hello` will run on IO thread and throw `BlockingOperationNotAllowedException` exception.
+xref:security-authorization.adoc#standard-security-annotations[Standard security annotations] on CDI beans are not supported on an I/O thread if a non-void secured method returns a value synchronously and proactive authentication is disabled because they need to access `SecurityIdentity`.
+
+In the following example, `HelloResource` and `HelloService` are defined.
+Any GET request to `/hello` will run on the I/O thread and throw a `BlockingOperationNotAllowedException` exception.
 
 There is more than one way to fix the example:
 
-* switch to a worker thread (annotate `hello` endpoint with `@Blocking`)
-* change `sayHello` method return type (use reactive or asynchronous data type)
-* arguably the safest way is to move `@RolesAllowed` annotation to the endpoint, as accessing `SecurityIdentity` from endpoint methods is never the blocking operation
```

Comment:
Making claims like "arguably the safest way..." could frustrate readers, especially if this claim is weak or becomes obsolete after the doc publish. I suggested a rephrase, but there might be a better way...

"This could be one of the safest ways because..."